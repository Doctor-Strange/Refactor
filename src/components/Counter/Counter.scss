@import "../../styles/colors";

.counter_container {
  *::selection {
    background-color: transparent;
  }
  .counter_box {
    margin-top: 1.2rem;
    margin-bottom: 2.5rem;
    .counter {
      display: inline-block;
      vertical-align: middle;
      p {
        display: inline-block;
        vertical-align: middle;
        width: 6rem;
        text-align: center;
        margin: 0 1.2rem;
        border: 1px solid $input_border;
        border-radius: 4px;
        line-height: 2;
      }
      svg {
        cursor: pointer;
        background-color: rgba($main__color, 0.15);
        padding: 0.7rem;
        border-radius: 4px;
        width: 1.6rem;
        height: 1.6rem;
        color: $main__color;
        display: inline-block;
        vertical-align: middle;
      }
    }
    .text_tail {
      display: inline-block;
      margin-right: 1rem;
    }
  }
}
