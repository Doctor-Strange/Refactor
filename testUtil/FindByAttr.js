

export const FindByAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`);
};
