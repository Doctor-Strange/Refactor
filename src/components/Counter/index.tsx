import React, { useState } from "react";
import "./Counter.scss";
import { IoMdAdd, IoMdRemove } from "react-icons/io";

const Counter = (props: ICounter) => {
  return (
    <div className="counter_container">
      <label>{props.label}</label>
      <div className="counter_box">
        <div className="counter">
          <IoMdAdd
            onClick={() => {
              console.log(props.value, props.max);

              if (props.value < props.max) props.AddTo();
            }}
          />
          <p>{props.value}</p>
          <IoMdRemove
            onClick={() => {
              if (props.value > props.min) props.reduceTo();
            }}
          />
        </div>
        <span className="text_tail">{props.text}</span>
      </div>
    </div>
  );
};

interface ICounter {
  max: number;
  min: number;
  AddTo: any;
  reduceTo: any;
  label: string;
  text: string;
  value: number;
}

export default Counter;
