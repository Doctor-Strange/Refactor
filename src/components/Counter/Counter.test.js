import React from "react";
import { mount } from "enzyme";
import { FindByAttr } from "../../../testUtil/FindByAttr";
import Counter from "./";

const setup = (props) => {
  return mount(<Counter {...props} />);
};

const AddToMock = jest.fn();
const reduceToMock = jest.fn();

const props = {
  max: 10,
  min: 0,
  AddTo: AddToMock,
  reduceTo: reduceToMock,
  label: "text",
  text: "placeHolder",
  value: 1,
};

describe("Counter component", () => {
  describe("snapshot", () => {
    test("/components/Counter", () => {
      const wrapper = setup(props);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("counter props", () => {
    const wrapper = setup(props);
    const label = FindByAttr(wrapper, "label");
    const tail = FindByAttr(wrapper, "tail");
    expect(label.text()).toEqual("text");
    expect(tail.text()).toEqual("placeHolder");
  });

  describe("counter mockfunction", () => {
    const wrapper = setup(props);
    const add = FindByAttr(wrapper, "add");
    const remove = FindByAttr(wrapper, "remove");

    add.first().simulate("click");
    expect(AddToMock).toHaveBeenCalledTimes(1);

    remove.first().simulate("click");
    expect(reduceToMock).toHaveBeenCalledTimes(1);
  });
});
