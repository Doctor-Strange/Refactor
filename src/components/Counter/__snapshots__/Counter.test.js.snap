// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component snapshot /components/Counter 1`] = `ReactWrapper {}`;
